test/val/val_id_test.cpp

```diff
@@ -537,9 +537,16 @@ enum Signed { kSigned, kUnsigned };
 // Creates an assembly snippet declaring OpTypeArray with the given length.
 string MakeArrayLength(const string& len, Signed isSigned, int width) {
   ostringstream ss;
-  ss << kGLSL450MemoryModel;
-  ss << " %t = OpTypeInt " << width << (isSigned == kSigned ? " 1" : " 0")
-     << " %l = OpConstant %t " << len << " %a = OpTypeArray %t %l";
+  ss << R"(
+    OpCapability Shader
+    OpCapability Linkage
```

Comment:
We might run into problems with this later because Vulkan doesn't allow Linkage.  But let's address that once it becomes a problem.  I know you did this because you don't want to have to name an entry point.